Index: quota-tools/Makefile.in
===================================================================
--- quota-tools.orig/Makefile.in	2015-05-25 16:07:56.715913765 +0800
+++ quota-tools/Makefile.in	2015-05-25 16:09:00.299913606 +0800
@@ -32,6 +32,7 @@
 locale_dir    = $(prefix)/share/locale
 sysconfdir    = @sysconfdir@
 datarootdir   = @datarootdir@
+man_2         = *.2
 
 RPCCLNTOBJS = rquota_xdr.o rquota_client.o rquota_clnt.o
 IOOBJS = quotaio.o quotaio_v1.o quotaio_v2.o quotaio_tree.o quotaio_rpc.o quotaio_xfs.o quotaio_meta.o quotaio_generic.o
@@ -100,7 +101,9 @@
 	-$(INSTALL) $(STRIP) -m $(DEF_SBIN_MODE) quota_nld $(ROOTDIR)$(sbindir)
 endif
 	-$(INSTALL) -m $(DEF_MAN_MODE) *.1 $(ROOTDIR)$(mandir)/man1
+ifeq ($(man_2),)
 	-$(INSTALL) -m $(DEF_MAN_MODE) *.2 $(ROOTDIR)$(mandir)/man2
+endif
 	-$(INSTALL) -m $(DEF_MAN_MODE) *.3 $(ROOTDIR)$(mandir)/man3
 	-$(INSTALL) -m $(DEF_MAN_MODE) *.8 $(ROOTDIR)$(mandir)/man8
 
