Upstream-Status: Pending

Subject: configure.ac: add valgrind header checking option

Signed-off-by: Chen qi <Qi.Chen@windriver.com>
---
 configure.ac | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 945adfc..dd4d7b2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1141,7 +1141,10 @@ AC_DEFINE_UNQUOTED(TELINIT, ["$TELINIT"], [Path to telinit])
 
 AC_SUBST(TELINIT)
 
-AC_CHECK_HEADERS_ONCE([valgrind/memcheck.h valgrind/valgrind.h])
+AC_ARG_ENABLE(valgrindheader, AS_HELP_STRING([--disable-valgrindheader], [disable valgrind header checking]))
+if test "x$enable_valgrindheader" != "xno"; then
+        AC_CHECK_HEADERS_ONCE([valgrind/memcheck.h valgrind/valgrind.h])
+fi
 
 # ------------------------------------------------------------------------------
 have_myhostname=no
-- 
1.9.1

