--- a/tmpfiles.d/etc.conf.m4.old	2016-01-07 23:45:37.882440317 -0800
+++ b/tmpfiles.d/etc.conf.m4	2016-01-07 23:45:43.942437718 -0800
@@ -7,7 +7,7 @@
 
 # See tmpfiles.d(5) for details
 
-L /etc/os-release - - - - ../usr/lib/os-release
+#L /etc/os-release - - - - ../usr/lib/os-release
 L /etc/localtime - - - - ../usr/share/zoneinfo/UTC
 L+ /etc/mtab - - - - ../proc/self/mounts
 m4_ifdef(`ENABLE_RESOLVED',
